### PR TITLE
handle undefined mtime

### PR DESCRIPTION
**Pull request by jace-roell**
## Proposed changes

- shifting focus to an open editor with a extender profile type that does not return `mtime` (i.e RSE) should reload the contents

**How to test**

- Open a USS file in VScode using a profile that does not return `mtime` (ex: RSE)
- Modify the opened USS outside of VScode (i.e, ssh, CLI, etc)
- Shifting focus to the VScode editor should fetch the new contents automatically

Note: if you do not have access to an API that does not return an `mtime` . Add the below code snippet after `USSFsProvider:269` in the `listFiles` function to test this behavior.

```
// TEST CHANGE: Strip out mtime and remove "." and ".." entries to simulate API that doesn't return them
            if (response.success && response.apiResponse?.items) {
                const originalItems = response.apiResponse.items;
                const itemsWithoutMtime = originalItems.map(item => {
                    const { mtime, ...itemWithoutMtime } = item;
                    return itemWithoutMtime;
                });
                const filteredItems = itemsWithoutMtime.filter((it): boolean => !/^\.{1,2}$/.test(it.name as string));
                response.apiResponse.items = filteredItems;
            }
```
## Release Notes

Milestone:

Changelog:

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**All checks must pass before this pull request is reviewed by the Zowe Explorer squad.**

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`
- [ ] New ZE APIs are tested with extender types that haven't adopted yet to determine breaking changes. Can use Zowe zFTP marketplace extension.

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [x] I have tested new functionality with the FTP extension and profile verifying no extender profile type breakages introduced
- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by JillieBeanSim — COMMENTED** [review]
thanks for working on this @jace-roell my extender profile's file in editor was updated with latest after a cli upload, no need for git-compare. ❤️

File: packages/zowe-explorer/src/trees/uss/UssFSProvider.ts [hunk 1]
```diff
@@ -132,11 +132,18 @@ export class UssFSProvider extends BaseProvider implements vscode.FileSystemProv
             if (fileResp.success) {
                 // Regardless of the resource type, it will be the first item in a successful response.
                 // When listing a folder, the folder's stats will be represented as the "." entry.
-                const newTime = dayjs((fileResp.apiResponse?.items ?? [])?.[0]?.mtime ?? entry.mtime).valueOf();
-                if (entry.mtime != newTime) {
-                    entry.mtime = newTime;
-                    // if the modification time has changed, invalidate the previous contents to signal to `readFile` that data needs to be fetched
+                const apiMtime = (fileResp.apiResponse?.items ?? [])?.[0]?.mtime;
+
+                if (apiMtime === undefined) {
```
Comment (traeok) [inline packages/zowe-explorer/src/trees/uss/UssFSProvider.ts]: (nit) we can shorten the expression and also cover null:

```suggestion
                if (apiMtime == null) {
```

**Review by traeok — APPROVED** [review]
LGTM, thanks Jace for the fix

**Review by JillieBeanSim — COMMENTED** [review]
Thanks for working on this @jace-roell I am still testing but wanted to mention that I think this issue is in the marketplace so we should include a changelog

File: packages/zowe-explorer/src/trees/uss/UssFSProvider.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ import {
     AuthHandler,
     IFileSystemEntry,
     ConflictViewSelection,
+    ConflictViewSelection,
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/trees/uss/UssFSProvider.ts]: this is a duplicate import

**Review by JillieBeanSim — DISMISSED** [review]
Thanks for the fix @jace-roell testing looks good, left a comment about a duplicate import and the changelog. once addressed will approve

**Comment by jace-roell** [comment]
> Thanks for the fix @jace-roell testing looks good, left a comment about a duplicate import and the changelog. once addressed will approve

Thank you, I have addressed the duplicate import. 👍 

When doing some more manual testing, I found another edge case with listing USS files with RSE. I opened an [issue](https://github.com/zowe/zowe-explorer-vscode/issues/4204) as I believe it can be addressed separately

**Review by zFernand0 — APPROVED** [review]
